pkg/flow/internal/worker/worker_pool_test.go

```diff
@@ -90,17 +91,18 @@ func TestWorkerPool(t *testing.T) {
 				tasksDone.Inc()
 			})
 			require.NoError(t, err)
+			defer func() { close(blockFirstTask) }()
 
 			// Wait for the first task to be running already and blocking the worker
 			<-firstTaskRunning
-			require.Equal(t, 0, pool.QueueSize())
+			require.Equal(t, 1, pool.QueueSize())
```

Comment:
The definition of QueueSize changed to also include the tasks that are currently executing. This makes things a bit simpler in the implementation, while still being able to perform its function: tell us when queue is building up.